### DISCUSSION TITLE
uc.click gives error: Invalid selector type: "<class 'seleniumbase.undetected.webelement.WebElement'>" Expecting a selector of type: "<class 'str'>" (string)!

### DISCUSSION THREAD

**Discussion by Vrackko** [seleniumbase/SeleniumBase#2760] [Q&A] [answered]
code:

`
try:
reklama1link = driver.find_elements(By.CLASS_NAME, "container-8f243bc68f55efcdfb1d0214c9b48c38__link")
print ("Reklama 1 links is found AND random link is selected")
except Exception as e:
print ("Reklama1 link wasnt found:", e)

    try:
        driver.is_element_visible(reklama1)
        print("Reklama1 is present")
    except Exception as e:
        print ("Reklama1 is not present", e)

   random_reklama1link = random.choice(reklama1link)

    driver.uc_click(random_reklama1link)
    print ("clicked")
`

error:
`
Traceback (most recent call last):
  File "c:\Users\HP\Desktop\No Com\proxy.py", line 120, in <module>
    driver.click(element)
  File "C:\Users\HP\Desktop\No Com\.venv\Lib\site-packages\seleniumbase\core\sb_driver.py", line 59, in click
    page_actions.click(self.driver, *args, **kwargs)
  File "C:\Users\HP\Desktop\No Com\.venv\Lib\site-packages\seleniumbase\fixtures\page_actions.py", line 1506, in click
    selector, by = page_utils.recalculate_selector(selector, by)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\HP\Desktop\No Com\.venv\Lib\site-packages\seleniumbase\fixtures\page_utils.py", line 92, in recalculate_selector
    raise Exception('Invalid selector type: "%s"\n%s' % (_type, msg))
Exception: Invalid selector type: "<class 'seleniumbase.undetected.webelement.WebElement'>"
Expecting a selector of type: "<class 'str'>" (string)!
`

Is there a way to uc.click html element that is stored in variable?
I tried with buttons in the same way, same error
p.s sorry if everything here is messy as i presented my code

**Comment by mdmintz** [comment, marked as answer]
The correct syntax is `driver.uc_click(selector)`.
You must pass a selector, not a WebElement.
`driver.find_elements` returns a list of WebElements.
